src/main/java/org/springframework/data/couchbase/core/convert/MappingCouchbaseConverter.java

```diff
@@ -503,7 +503,7 @@ protected void writeInternal(final Object source, final CouchbaseDocument target
 		final TreeMap<Integer, String> suffixes = new TreeMap<>();
 		final TreeMap<Integer, String> idAttributes = new TreeMap<>();
 
-		target.setExpiration(entity.getExpiry());
+		target.setExpiration((int)(entity.getExpiryDuration().getSeconds()));
 
```

Comment:
use explicit getExpiryDuration() as the other getExpiry() is sometimes duration and sometimes instant.